app/components/primer/alpha/tooltip.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+module Primer
+  module Alpha
+    # `Tooltip` is a wrapper component that will apply a tooltip to the provided content.
+    #
+    # The element that triggers the tooltip must be rendered through the `trigger` slot.
+    # If the trigger element is a view component, use the `trigger_component` slot.
+    class Tooltip < Primer::Component
+
+      TYPE_FALLBACK = :describe
+      TYPE_OPTIONS = [TYPE_FALLBACK, :label].freeze
+
+      TRIGGER_ELEMENT_FALLBACK_TAG = :button
+      TRIGGER_ELEMENT_TAG_OPTIONS = [
+        TRIGGER_ELEMENT_FALLBACK_TAG,
+        :a,
+        :summary,
+        :"md-header",
+        :"md-bold",
+        :"md-italic",
+        :"md-quote",
+        :"md-code",
+        :"md-link",
+        :"md-ordered-list",
+        :"md-task-list",
+        :"md-unordered-list"
+      ].freeze
+
+      DIRECTION_DEFAULT = :n
+      MULTILINE_DEFAULT = false
+
+      DIRECTION_OPTIONS = [DIRECTION_DEFAULT] + %i[
+        n
+        w
+        e
+        s
+        ne
+        se
+        sw
+        nw
+      ]
+
+      # The element that triggers the tooltip
+      #
+      # @param tag [Symbol] <%= one_of(Primer::Alpha::Tooltip::TRIGGER_ELEMENT_TAG_OPTIONS) %>
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :trigger, lambda { |tag:, **system_arguments|
+        system_arguments[:tag] = fetch_or_fallback(TRIGGER_ELEMENT_TAG_OPTIONS, tag, TRIGGER_ELEMENT_FALLBACK_TAG)
+        @type == :describe ? system_arguments[:"aria-describedby"] = @tooltip_id : system_arguments[:"aria-labelledby"] = @tooltip_id
+        Primer::BaseComponent.new(**system_arguments)
+      }
+
+      # The component that triggers the tooltip
+      #
+      # @param klass [Class] View component that triggers the tooltip.
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_one :trigger_component, lambda { |klass:, **system_arguments|
```

Comment:
Using slots feels a teeny bit awkward, but I don't see how else we can build in restrictions for what elements/components can render a tooltip.